rdr_service/dao/study_nph_dao.py

```diff
@@ -171,6 +176,14 @@ def timepoint_exist(order: Namespace, visit_type: StudyCategory, session) -> Tup
                 return True, result
         return False, None
 
+    def get_study_category(self, study_category_id: int) -> StudyCategory:
+        with self.session() as session:
+            return session.query(StudyCategory).get(study_category_id)
+
+    def get_parent_study_category(self, study_category_id: int) -> StudyCategory:
+        study_category = self.get_study_category(study_category_id)
+        return self.get_study_category(study_category.parent_id)
```

Comment:
Is there a better way to handle this since it making two consecutive db calls?

Reply:
We can prevent memoize SELECT calls to the db to fetch the same StudyCategory object by caching it in the RAM using LRU Cache of fixed size. I am not sure how to do that for a non `staticmethod`.